fix tsnr_afterburner exposures SURVEY

sbailey opened the pull request:
This PR fixes the tsnr_afterburner exposures vs. tiles SURVEY inconsistency reported in #1649.  Admittedly it does this by adding another special case check in the exposures logic rather than attempting a deeper refactor to reduce technical debt.

Other updates:

Adds a PROGRAM column, which is the output of desispec.io.meta.faflavor2program(FAFLAVOR), standardizing the many sv1 flavors/faprgrm into the standardized dark/bright/backup/other to match current usage and to match how the healpix redshifts are grouped.  For sv3 and main tiles, FAPRGRM==PROGRAM, but for sv1 tiles this should make interpreting the healpix structure easier.

It also fixes an MPI logic bug if size>nights (the same night was proceeded by multiple ranks, leading to race conditions), and renames `--multinode` to `--mpi` for consistency with other desispec scripts.  The MPI route still requires --recompute which I haven't tried to fix here.

Lastly, if --tile-completeness is a fits file, it also writes a .csv version, mirroring the behavior of --outfile for the exposures file which writes both a fits and a csv version.

This PR does not try to fix the GOALTIME inconsistency, nor the EFFTIME_ETC.

Example outputs are in /global/cfs/cdirs/desi/users/sjbailey/dev/fuji/exp.fits and tiles.fits, generated with
```
time desi_tsnr_afterburner -o exp.fits -t tiles.fits --aux /global/cfs/cdirs/desi/survey/observations/SV1/sv1-tiles.fits --gfa-proc-dir /global/cfs/cdirs/desi/survey/GFA/ --nproc 16
```
[45 minutes]
Note that I did *not* include --compute-skymags in that run

akremin reviewed (approved):
The code looks good.

I compare the two `csv` versions of the example tables. Among the columns in common I see agreement on all tiles entries.

Columns compared for each tile: `['TILEID', 'TILERA', 'TILEDEC', 'SURVEY', 'FAPRGRM', 'FAFLAVOR', 'GOALTIME', 'GOALTYPE', 'MINTFRAC']`

Code used:
```
from astropy.table import Table 
import numpy as np 
exps = Table.read("/global/cfs/cdirs/desi/users/sjbailey/dev/fuji/exp.csv")
tiles =  Table.read("/global/cfs/cdirs/desi/users/sjbailey/dev/fuji/tiles.csv")
overlap = [col for col in exps.colnames if col in tiles.colnames and 'EFFTIME' not in col and 'EXPTIME' not in col] 
for i,tile in enumerate(tiles['TILEID']): 
     itterexps = exps[exps['TILEID'] == tile] 
     ittertile = tiles[i] 
      
     for col in overlap: 
         eversion = np.unique(itterexps[col].data) 
         tversion = np.atleast_1d(ittertile[col]) 
         if not np.all(eversion == tversion): 
             print(f"{tile}\t{col}\t{eversion}\t{tversion}") 
```

The only things printed are related to `GOALTIME`, as expected since it is acknowledged in the PR summary that that issue is unresolved.

sbailey commented:
Post review update:
  * added EXTNAME=TILE_COMPLETENESs to tiles.fits table
  * added missing PROGRAM column to tiles table (it was in exposures, but not tiles)
